src/service/viewer-impl.js

```diff
@@ -392,6 +392,41 @@ export class Viewer {
   }
 
   /**
+   * Returns the "referrer" URL that can be optionally customized by
+   * the viewer.
+   * @return {string}
+   */
+  getReferrerUrl() {
+    // If not embedded in a IFrame always return the document's referrer.
+    if (!this.isEmbedded()) {
+      return this.win.document.referrer;
+    }
+
+    // Viewer override.
+    if (this.params_['referrer']) {
+      return this.params_['referrer'];
```

Comment:
We should only trust this for trusted viewers. That is a concept we need to introduce.

Basically if the referrer/ancestorOrigins is google.com, we'd do this.